lib/modules/ens/index.js

```diff
@@ -14,6 +14,13 @@ class ENS {
     this.registration = this.namesConfig.register || {};
     this.embark = embark;
 
+    if (this.namesConfig === {} ||
+      this.namesConfig.enabled !== true ||
+      this.namesConfig.available_providers.indexOf('ens') < 0 ||
+      this.namesConfig.provider !== 'ens') {
```

Comment:
what if the provider is `ipns` but `ens` is part of the available providers.
Should it still skip everything?

Reply:
Are the ENS contracts and/or ENS functions needed for IPNS?
Sorry if it's a stupid question, I'm still unsure what is IPNS

Reply:
IPNS is the equivalent but for IPFS.

@alaibe this is behaviour is correct because this is the ens module, so it only affect the ens stuff.

Reply:
If you have this config
```{enabled: true, available_providers: ['ens', 'ipns'], provider: 'ipns'}```

Both providers are available. According to your conditions, the ens contract will not be deployed because of `this.namesConfig.provider !== 'ens'` but they should

So removing `this.namesConfig.provider !== 'ens'` should make it works
What do you think?